packages/react-components/react-datepicker-compat/src/components/DatePicker/DatePicker.cy.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
+import { mount as mountBase } from '@cypress/react';
+
+import { FluentProvider } from '@fluentui/react-provider';
+import { teamsLightTheme } from '@fluentui/react-theme';
+
+import { DatePicker } from './DatePicker';
+
+const mount = (element: JSX.Element) => {
+  mountBase(<FluentProvider theme={teamsLightTheme}>{element}</FluentProvider>);
+};
+
+const inputSelector = '[role="combobox"]';
+const popoverSelector = '[role="dialog"]';
+const fieldErrorMessageSelector = '[role=alert]';
+
+describe('DatePicker', () => {
+  it('opens a default datepicker', () => {
+    mount(<DatePicker />);
+    cy.get(inputSelector).click().get(popoverSelector).should('be.visible');
+  });
+
+  it('should not open a datepicker when disabled', () => {
+    mount(<DatePicker disabled />);
+    // Force is needed because otherwise Cypress throws an error
+    cy.get(inputSelector).click({ force: true }).get(popoverSelector).should('not.exist');
+  });
+
+  it('should render DatePicker and calloutId must exist in the DOM when isDatePickerShown is set', () => {
+    mount(<DatePicker />);
+    cy.get(inputSelector).click();
+
+    cy.get('body').find('[aria-owns]').should('exist');
+  });
+
+  it('should clear error message when required input has date text and allowTextInput is true', () => {
+    mount(<DatePicker isRequired allowTextInput />);
+
+    // Open DatePicker and dismiss
+    cy.get(inputSelector).click().get('body').click('bottomRight').get(fieldErrorMessageSelector).should('exist');
```

Comment:
You should probably separate this into 2 statements for clarity

```suggestion
    cy.get(inputSelector).click().get('body').click('bottomRight');
    cy.get(fieldErrorMessageSelector).should('exist');
```